t/18-qemu.t

```diff
@@ -391,17 +374,22 @@ subtest DriveDevice => sub {
 subtest 'relative assets' => sub {
     $vars{$_} = "Core-7.2.iso" for qw(ISO ISO_1 HDD_1 UEFI_PFLASH_VARS);
     symlink("$Bin/data/Core-7.2.iso", "./Core-7.2.iso");
-    $proc = OpenQA::Qemu::Proc->new()
-      ->_static_params(['-foo'])
-      ->qemu_bin('qemu-kvm')
-      ->qemu_img_bin('qemu-img')
-      ->configure_blockdevs('disk', 'raid', \%vars)
-      ->configure_pflash(\%vars);
+    $proc = qemu_proc('-foo', \%vars);
     my @gcmdl = $proc->blockdev_conf->gen_qemu_img_cmdlines();
     @cmdl = map { [qw(create -f qcow2 -b), "$dir/Core-7.2.iso", "raid/$_-overlay0", 11116544] } qw(hd0 cd0 cd1);
     push @cmdl, ["create", "-f", "qcow2", "-b", "$Bin/data/uefi-code.bin", "raid/pflash-code-overlay0", 1966080];
     push @cmdl, ["create", "-f", "qcow2", "-b", "$dir/Core-7.2.iso",       "raid/pflash-vars-overlay0", 11116544];
     is_deeply(\@gcmdl, \@cmdl, 'find the asset real path');
+};
+
+subtest 'qemu was killed due to the system being out of memory' => sub {
+    $bmwqemu::scriptdir = "$Bin/fake/tests";
```

Comment:
This is an interesting approach but it feels like bit of an overhead of having a separate fake script. Also feels like cheating on the code coverage infrastructure. How about keeping the approach of having a single-line sub that can be mocked? In that single-line sub include the call to the external script and mark the sub with `# uncoverable statement`. Maybe better if that works: Mock `system`, e.g. `*CORE::system` or something

Reply:
I tried Mock system in test code, but it didn't work.  So I changed that to a single-line sub.